session/session.go

```diff
@@ -1466,7 +1466,7 @@ func (s *session) Parse(ctx context.Context, sql string) ([]ast.StmtNode, error)
 
 // ParseWithParams parses a query string, with arguments, to raw ast.StmtNode.
 // Note that it will not do escaping if no variable arguments are passed.
-func (s *session) ParseWithParams(ctx context.Context, sql string, args ...interface{}) (ast.StmtNode, error) {
+func (s *session) ParseWithParams(ctx context.Context, ignoreClientCharsetAndCollation bool, sql string, args ...interface{}) (ast.StmtNode, error) {
```

Comment:
In what situation the ignoreClientCharsetAndCollation is false? Any test for it?

Reply:
I just translate the old change to make it right.

@Defined2014 What situation do we ignore the client's charset and collation ?

Reply:
https://github.com/pingcap/tidb/pull/30843#discussion_r772128565

Reply:
For now, every SQLs should ignore the client's charset and collation.

Reply:
If so, why do you provide this argument if it's useless?

Reply:
@xhebox PTAL

Reply:
For all internal SQLs, it is true. SQLs from clients should be true. I recommeded to make the param from `ignoreClientCharsetAndCollation ` to `isInternal`..

Reply:
Can we remove the code about getting the client's charset and collation?

Reply:
> For all internal SQLs, it is true. SQLs from clients should be false. I recommeded to make the param from `ignoreClientCharsetAndCollation ` to `isInternal`, instead of just ignoring charset. @Defined2014 @hawkingrei

In fact, ```isInternal``` is ```s.sessionVars.InRestrictedSQL```. It will cause side effects about telemetry. #30843 just don't transform charset in internal sql. why does it change InRestrictedSQL value and recovery value after execute.

Reply:
InRestrictedSQL should be false for user sqls. SO only when we executing internal SQLs, it is setting to true.

Reply:
Fixed